paraglob.cpp

```diff
@@ -0,0 +1,111 @@
+#include "paraglob.h"
+
+Paraglob::Paraglob(const std::vector<std::string> &patterns): Paraglob() {
+  for (std::string pattern : patterns) {
+    this->add(pattern);
+  }
+  this->compile();
+}
+
+void Paraglob::add(std::string pattern) {
+  // Get the meta words
+  std::vector<std::string> m_words = get_meta_words(pattern);
+  // Put them away
+  if (m_words.size() == 0) {
+    this->single_wildcards.push_back(pattern);
+  }
+  for (std::string meta_word : m_words) {
+    AhoCorasickPlus::EnumReturnStatus status;
+    AhoCorasickPlus::PatternId patId = this->meta_words.size();
+
+    status = this->my_ac.addPattern(meta_word, patId);
+    if (status != AhoCorasickPlus::RETURNSTATUS_SUCCESS &&
+      status != AhoCorasickPlus::RETURNSTATUS_DUPLICATE_PATTERN) {
+      std::cout << "Failed to add: " << meta_word << std::endl;
```

Comment:
Should return an error indicator instead of printing to stdout, so that the caller can handle as appropriate.